src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/controllers/v1/TransactionsController.kt

```diff
@@ -0,0 +1,106 @@
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.services.GetTransactionsForPersonService
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.services.internal.AuditService
+import java.time.LocalDate
+
+@RestController
+@RequestMapping("/v1/prison/{prisonId}/prisoners/{hmppsId}/transactions/{accountCode}")
+class TransactionsController(
+  @Autowired val auditService: AuditService,
+  @Autowired val getTransactionsForPersonService: GetTransactionsForPersonService,
+) {
+  @Operation(
+    summary = "Returns all transactions for a prisoner associated with an account code that they have at a prison.",
+    description = "<b>Applicable filters</b>: <ul><li>prisons</li></ul>",
+    responses = [
+      ApiResponse(responseCode = "200", useReturnTypeSchema = true, description = "Successfully found a prisoner's transactions."),
+      ApiResponse(
+        responseCode = "400",
+        description = "The request data has an invalid format or the prisoner does hot have transactions at the specified prison.",
+        content = [
+          Content(
+            schema =
+              io.swagger.v3.oas.annotations.media
+                .Schema(ref = "#/components/schemas/BadRequest"),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        content = [
+          Content(
+            schema =
+              io.swagger.v3.oas.annotations.media
+                .Schema(ref = "#/components/schemas/PersonNotFound"),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "500",
+        content = [
+          Content(
+            schema =
+              io.swagger.v3.oas.annotations.media
+                .Schema(ref = "#/components/schemas/InternalServerError"),
+          ),
+        ],
+      ),
+    ],
+  )
+  @GetMapping()
+  fun getTransactionsByAccountCode(
+    @PathVariable hmppsId: String,
+    @PathVariable prisonId: String,
+    @PathVariable accountCode: String,
+    @RequestAttribute filters: ConsumerFilters?,
+    @Parameter(description = "Start date for transactions (defaults to today if not supplied)") @RequestParam(required = false, name = "from_date") fromDate: String?,
+    @Parameter(description = "To date for transactions (defaults to today if not supplied)") @RequestParam(required = false, name = "to_date") toDate: String?,
+  ): DataResponse<Transactions?> {
+    var startDate = LocalDate.now().toString()
+    var endDate = LocalDate.now().toString()
+
+    if (fromDate == null && toDate != null || toDate == null && fromDate != null) {
+      throw ValidationException("Both fromDate and toDate must be supplied if one is populated")
+    }
+
+    if (fromDate != null && toDate != null) {
+      startDate = fromDate
+      endDate = toDate
+    }
+
+    if (accountCode !in listOf("spends", "savings", "cash")) {
+      throw ValidationException("Account code must either be 'spends', 'savings', or 'cash'")
+    }
```

Comment:
We should probably be aiming to do as much of this validation in the service as possible